File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def oddEvenList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        arr = []
        temp = head
        if temp == None or temp.next == None:
            return temp
        while temp!= None and temp.next != None:
            arr.append(temp.val)
            temp = temp.next.next
        if temp: arr.append(temp.val)
        temp = head.next
        while temp!=None and temp.next != None:
            arr.append(temp.val)
            temp = temp.next.next
        if temp: arr.append(temp.val)
        temp = head
        for i in range(len(arr)):
            if i == 0: node = temp
            temp.val = arr[i]
            temp = temp.next
        return node
            

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function(head) {
    if (!head) return null
    let first = head
    let second = head.next
    let frst_sec = second
    while (second && second.next) {
        first.next = first.next.next
        second.next = second.next.next
        first = first.next
        second = second.next
    }
    first.next = frst_sec
    return head
};